src/bindings/c/include/openvino/c/deprecated.h

```diff
@@ -60,6 +60,11 @@
 #    define OPENVINO_ENUM_DEPRECATED(msg)
 #endif
 
+#define IE_1_0_DEPRECATED                                                                                    \
+    OPENVINO_DEPRECATED("The Inference Engine API is deprecated and will be removed in the 2024.0 release. " \
+                        "For instructions on transitioning to the new API, please refer to "                 \
+                        "https://docs.openvino.ai/latest/openvino_2_0_transition_guide.html")
```

Comment:
please, move this declaration to ie_c_api.h, because we don't need to mix headers for new and old APIs.